File: 백준/Silver/2108. 통계학/통계학.py
from collections import Counter
import sys
input = sys.stdin.readline

n = int(input())
array = []
for _ in range(n):
    num = int(input())
    array.append(num)

for i in range(4):
    if i == 0:
        print(round(sum(array)/n))
    elif i == 1:
        array.sort()
        print(array[n//2])
    elif i == 2:
        cnt = Counter(array).most_common(2)

        if len(array) > 1:
            if cnt[0][1] == cnt[1][1]:
                print(cnt[1][0])
            else:
                print(cnt[0][0])
        else:
            print(cnt[0][0])
    elif i == 3:
        print(max(array) - min(array))
    elif i == 4:
        print(max(array) - min(array))

File: 백준/Silver/2108. 통계학/통계학.java
import java.util.*;
import java.lang.*;
import java.io.*;

class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int[] arr = new int[n];
        
        for(int i=0; i<n; i++) {
            int a = Integer.parseInt(br.readLine());
            arr[i] = a;
        }

        int sum = 0;
        for(int a : arr) {
            sum += a;
        }

        int average = Math.round((float) sum / n);

        Arrays.sort(arr);

        boolean check = false;
        int count = 0;
        int max = -1;
        int mod = arr[0];
        for(int i=0; i<n-1; i++){
            if(arr[i] == arr[i+1]) {
                count++;
            } else {
                count = 0;
            }

            if(max < count) {
                max = count;
                mod = arr[i];
                check = true;
            } else if(max == count && check == true) {
                mod = arr[i];
                check = false;
            }
        }
         
        System.out.println(average);
        System.out.println(arr[(n-1)/2]);
        System.out.println(mod);
        System.out.println(arr[n-1] - arr[0]);
    }
}